test_conformance/relationals/test_shuffles.cpp

```diff
@@ -260,33 +263,48 @@ char * get_order_name( ExplicitType vecType, size_t inVecSize, size_t outVecSize
     return orderName;
 }
 
-void    print_hex_mem_dump( const unsigned char *inDataPtr, const unsigned char * inDataPtr2, const unsigned char *expected, const unsigned char *outDataPtr, size_t inVecSize, size_t outVecSize, size_t typeSize )
+void print_hex_mem_dump(const unsigned char *inDataPtr,
+                        const unsigned char *inDataPtr2,
+                        const unsigned char *expected,
+                        const unsigned char *outDataPtr, size_t inVecSize,
+                        size_t outVecSize, size_t typeSize)
 {
-    char error [4096] = "";
-    strcat(error, "      Source: ");
-    for( unsigned int j = 0; j < inVecSize * typeSize; j++ )
+    auto print_hex_byte = [](unsigned char v) {
+        // Use a new stream to avoid manipulating state of outer stream.
+        std::ostringstream ss;
+        ss << std::setfill('0') << std::setw(2) << std::right << std::hex << +v;
```

Comment:
`+v`?  Is that intentional?

Reply:
Does this promote its type, somehow, to prevent it from being appended to the string as an actual character?  (Apologies if that is so - it's not something I've come across - so I'm wondering if that's the generally accepted idiom for this?)

Reply:
> Does this promote its type, somehow, to prevent it from being appended to the string as an actual character?

Correct, the unary `+` operand is subject to implicit integer promotion as the rank of `unsigned char` is less than the rank of `int`.  I believe it's a fairly common idiom when dealing with output streams and `char` types (and in general, one of the very few practical uses of unary `+`: to force integer promotion).

Reply:
Thanks!  If that's the common idiom, then the change looks good to me.

(I would raise a bit of an eyebrow at iostream, here – in C, _char_ is a "byte" of unspecified signedness (i.e. usually 8 bits) and as I understood it, character constants are actually of type _int_ – so speaking as someone more used to printf, this seems highly questionable – although I suppose if you wanted an 8-bit integer and not a _character_, you'd use _signed char_ or _unsigned char_ (or _int8_t_ or _uint8_t_) – but then I suppose these get interpreted by iostream as characters too.  Oh well...)